configstack/stack.go

```diff
@@ -25,6 +26,40 @@ func (stack *Stack) String() string {
 	return fmt.Sprintf("Stack at %s:\n%s", stack.Path, strings.Join(modules, "\n"))
 }
 
+// Plan execute plan in the given stack in their specified order.
+func (stack *Stack) Plan(terragruntOptions *options.TerragruntOptions) error {
+	stack.setTerraformCommand([]string{"plan"})
+
+	// We capture the out stream for each module
+	errorStreams := make([]bytes.Buffer, len(stack.Modules))
+	for n, module := range stack.Modules {
+		module.TerragruntOptions.ErrWriter = &errorStreams[n]
+	}
+
+	// Start the plan on each module
+	err := RunModules(stack.Modules)
+
+	// We inspect the error streams to give an explicit message if the plan failed because there were references to
+	// remote states. `terraform plan` will fail if it tries to access remote state from dependencies and the plan
+	// has never been applied on the dependency.
+	for n, o := range errorStreams {
```

Comment:
`i` is a more common name for an index variable. `errorStream` is probably an easier to understand name than `o`.